flintrock/ec2.py

```diff
@@ -57,33 +58,47 @@ def __init__(
             vpc_id: str,
             master_instance: 'boto3.resources.factory.ec2.Instance',
             slave_instances: "List[boto3.resources.factory.ec2.Instance]",
+            use_private_vpc,
```

Comment:
> Do we include a subnet_is_private method that queries the subnet to determine if it private?

I think `subnet_is_private` should be a `@property`.

> In the case we make this change, get_cluster, get_clusters and _compose_cluster also need the subnet_id.

Do you mean as an input parameter? Ahhhh... I see what you were saying earlier now.

I think the signatures of these methods should stay the same (we don't need the subnet to locate a cluster), but when they return a cluster object it should have the `subnet_id` defined.

We will need to query an instance (e.g. the master) to get that information, like you were suggesting earlier. My apologies.